index.js

```diff
@@ -31,6 +34,73 @@ async function run() {
+        }
+
+        // Get the external secrets
+        if (secrets.toLowerCase() != 'true') {
+            const secretsPath = path.isAbsolute(secrets) ?
+              secrets :
+              path.join(process.env.GITHUB_WORKSPACE, secrets);
+            try {
+                // FIXME Workaround for existSync since tests weren't passing with existSync.
+                const secretsContent = require(secretsPath);
+                if (!Array.isArray(secretsContent.secrets)) {
+                    throw new Error('Invalid external secrets file: secrets section must be an array')
+                }
+                if (containerDef.secrets) {
+                    // Some secrets already exist in task definition.
+                    if (!Array.isArray(containerDef.secrets)) {
+                        throw new Error('Invalid task definition format: secrets section must be an array')
+                    }
+                    for (var i = 0, len = secretsContent.secrets.length; i < len; i++) {
+                        containerDef.secrets.push(secretsContent.secrets[i])
+                    }
+                } else {
+                    containerDef.secrets = secretsContent.secrets
+                }
+            } catch (e) {
+                if (e.code == 'MODULE_NOT_FOUND') {
+                    throw new Error(`Secrets file does not exist: ${secrets}`);
+                } else {
+                    throw new Error(e.message)
+                }
+            }
+        } else {
+            // Check the task definition if secrets section exists
+            // It at least must be empty.
+            if (!Array.isArray(containerDef.secrets)) {
+                throw new Error('Invalid task definition format: secrets section must be an array')
+            }
+        }
+
+        // Loop through array of, hopefully, dictionaries in task def
+        containerDef.secrets.forEach(element => {
+            const valueFrom = element.valueFrom.split(":")
+            if (valueFrom.length != 2) {
+                // Detailed error since we're looping through.
+                throw new Error(`Invalid task definition: valueFrom format must be in the form of <service>:<path of variable>. Errored: ${valueFrom}`);
+            }
+            if (!['ssm', 'secretsmanager'].includes(valueFrom[0])) {
+                throw new Error(`Invalid task definition: valueFrom must have prefix ssm or secretsmanager, not ${valueFrom[0]}`)
+            }
+            if (valueFrom[1].charAt(0) != '/') {
+                valueFrom[1] = "/".concat(valueFrom[1])
+            }
+            element.valueFrom = `arn:aws:${valueFrom[0]}:${regionName}:${accountID}:${(valueFrom[0] == 'ssm') ? 'parameter' : 'secret'}${valueFrom[1]}`
+
+            // Set executionRoleArn to use secrets section
+            taskDefContents.executionRoleArn = `arn:aws:iam::${accountID}:role/ecsTaskExecutionRole`
```

Comment:
FYI the execution role may not always have this name, it's valid to use a custom one.

Reply:
Huh the more you know... I think the generic merge might be necessary.